tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -3105,6 +3107,68 @@ int BPF_KPROBE(trace_mprotect_alert)
     return 0;
 }
 
+#define BPF_OBJ_NAME_LEN 16U
+
+SEC("kprobe/security_bpf")
+int BPF_KPROBE(trace_security_bpf)
+{
+    if (!should_trace())
+        return 0;
+
+    buf_t *submit_p = get_buf(SUBMIT_BUF_IDX);
+    if (submit_p == NULL)
+        return 0;
+
+    set_buf_off(SUBMIT_BUF_IDX, sizeof(context_t));
+
+    context_t context = init_and_save_context(ctx, submit_p, SECURITY_BPF, 1 /*argnum*/, 0 /*ret*/);
+
+    u64 *tags = bpf_map_lookup_elem(&params_names_map, &context.eventid);
+    if (!tags)
+        return -1;
+
+    /* 1st argument == cmd (int) */
+    int cmd = (int)PT_REGS_PARM1(ctx);
+
+    save_to_submit_buf(submit_p, (void *)&cmd, sizeof(int), INT_T, DEC_ARG(0, *tags));
+
+    events_perf_submit(ctx);
+    return 0;
+};
+
+SEC("kprobe/security_bpf_map")
+int BPF_KPROBE(trace_security_bpf_map)
+{
+    if (!should_trace())
+        return 0;
+
+    buf_t *submit_p = get_buf(SUBMIT_BUF_IDX);
+    if (submit_p == NULL)
+        return 0;
+
+    set_buf_off(SUBMIT_BUF_IDX, sizeof(context_t));
+
+    context_t context = init_and_save_context(ctx, submit_p, SECURITY_BPF_MAP, 2 /*argnum*/, 0 /*ret*/);
+
+    u64 *tags = bpf_map_lookup_elem(&params_names_map, &context.eventid);
+    if (!tags)
+        return -1;
+
+    struct bpf_map *map;
+    char map_name[BPF_OBJ_NAME_LEN];
+
+    map = (struct bpf_map *)PT_REGS_PARM1(ctx);
+    bpf_probe_read_kernel_str(&map_name, BPF_OBJ_NAME_LEN, (void *)&map->name);
```

Comment:
nit: save_str_to_buf already reads the string from the given pointer, so I don't think this is necessary (map->name can be given directly to save_str_to_buf)

Reply:
Yes, it does. And also a leftover from me reading map_name for != '\0' as I had before. Fixing it.

Reply:
Alright, fixed it. Tested, all good. Minor comment line change in security_bpf krpobe so both are the same. Pushing it.